### DISCUSSION TITLE
About tiddlywiki language switching

### DISCUSSION THREAD

**Discussion by oeyoews** [TiddlyWiki/TiddlyWiki5#8526] [General]
Like any other multilingual website, I think it is important to put the language switch in a place where users can see it at first glance. In addition, I think it is more appropriate to use the corresponding language in the drop-down box instead of using English. Here is an example in Chinese. @Jermolene 

![image](https://github.com/user-attachments/assets/27ef5f74-5975-4879-a2e6-ee44d5a37051)

**Comment by pmario** [comment]
> ... I think it is more appropriate to use the corresponding language in the drop-down box instead of using English.

I think this text has to be fixed by translators, not TiddlyWiki. As you can see the German version says: Deutsch (Deutschland) and Deutsch (Österreich) because I did name them that way in the translation files.

**Comment by Jermolene** [comment]
Hi @oeyoews I agree with both points: (a) the language switcher should optionally appear next to the chevron that opens/closes the sidebar and (b) the language switcher should display the localised name of the language (I note that cs-CZ already localises the name of the language).

To accomplish (b) will require some care but should be feasible with some research.

A related point is that my understanding is that it is not recommended to use flags to represent languages (see [this article](https://localizejs.com/articles/why-using-flag-icons-can-confuse-your-users) for an example of the arguments against the practice). I wonder if we should change to using the ISO language codes in the switch (eg "en-GB")

**Reply by pmario** [reply]
I'm OK if flags are not used at tiddlywiki.com if we want to be politically neutral.

But I also want to have the freedom to represent my country and language using a flag, for exactly the reason why they are "demonised" in the article.

**Comment by Leilei332** [comment]
I find it strange that some language plugin's description uses native language while some not. I think the `description` field should be modified to use the native language.

![图片](https://github.com/user-attachments/assets/bcbe55e4-9e5c-439f-8218-e996c6962707)

**Reply by Jermolene** [reply]
Hi @Leilei332 it would be great to fix things to be consistent

**Comment by Leilei332** [comment]
There exists a problem for translating `Chinese (Traditional)`: Whether we should translate it into 繁體中文 or 正體中文 ?
